### PR TITLE
cmd/operator-sdk/run: remove legacy run subcommand

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**
Remove the legacy `run` subcommand. This also incidentally removes support for `run local` for legacy Ansible and Helm projects

**Motivation for the change:**
To remove legacy code in preparation for 1.0 release.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by hasbro17** [comment]
We have some links in the CHANGELOG referencing `./website/content/en/docs/cli/operator-sdk_run.md`.
https://github.com/operator-framework/operator-sdk/blob/master/CHANGELOG.md#v0150

We could update those to be non-relative links https://v0-18-x.sdk.operatorframework.io/docs/cli/operator-sdk_run/

Or just use the git sha if we don't want to point to the website in such an older release.

**Comment by hasbro17** [comment]
Ah this get's rid of `run packagemanifests` as well. Great, I don't have to push my PR now :)

**Comment by joelanford** [comment]
@hasbro17 :rofl: Sorry, I got overzealous.

I updated the links to point to the docs as they existed at the time when they were referenced (so to our old github docs at the v0.15.0 tag)

**Comment by joelanford** [comment]
I also went ahead and removed legacy cleanup command support with the same changes that @estroz suggested.

**Comment by hasbro17** [comment]
@estroz We can remove the following internal pkgs/code for legacy `generate packagemanifests` right?
https://github.com/operator-framework/operator-sdk/blob/master/internal/generate/olm-catalog/package_manifest.go
https://github.com/operator-framework/operator-sdk/blob/master/internal/generate/olm-catalog/package_manifest_test.go

But I can do a follow up for that.

**Comment by hasbro17** [comment]
Per offline discussion, this needs a CHANGELOG fragment so we can track and summarize all CLI changes in the `1.0.0` release notes.

**Comment by joelanford** [comment]
If CI passes on this commit, let's just commit and I'll do a fragment follow-up. I need to do one for #3385 as well.

File: changelog/fragments/rm-legacy-run.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Remove legacy run amd cleanup commands
```
Comment (tengqm) [inline changelog/fragments/rm-legacy-run.yaml:5]: amd -> and

**Pull request merged**